From f88c4b6088bd6b295bb343da461ac23e89890f35 Mon Sep 17 00:00:00 2001
From: "Baoli.Zhang" <baoli.zhang@intel.com>
Date: Mon, 31 May 2021 10:40:05 +0800
Subject: [PATCH] Enable existing kernel source code in local directory

Signed-off-by: Baoli.Zhang <baoli.zhang@intel.com>
---
 debian/rules | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/debian/rules b/debian/rules
index 3659e5b2b..00994f0ce 100755
--- a/debian/rules
+++ b/debian/rules
@@ -62,6 +62,15 @@ else
 	tar -C ../orig -xaf $(TAR_ORIG)
 endif
 
+DIR_KERNEL = ../kernel/
+
+kernel: $(DIR_KERNEL)
+	rsync --delete --exclude /debian --exclude .svk --exclude .svn --exclude .git --link-dest=$(DIR_KERNEL)/ -a $(DIR_KERNEL)/ .
+	QUILT_PATCHES='$(CURDIR)/debian/patches' QUILT_PC=.pc quilt push --quiltrc - -a -q
+
+$(DIR_KERNEL):
+	if [ ! -d $(DIR_KERNEL) ]; then $(error Cannot find kernel source at $(DIR_KERNEL)); fi
+
 CLEAN_PATTERNS := $(BUILD_DIR) $(STAMPS_DIR) debian/lib/python/debian_linux/*.pyc debian/lib/python/debian_linux/__pycache__ $$(find debian -maxdepth 1 -type d -name 'linux-*') debian/*-modules-*-di* debian/kernel-image-*-di* debian/*-tmp debian/*.substvars
 
 maintainerclean:
-- 
2.25.1

